### PR TITLE
nodejs_20: init at 20.0.0

### PR DESCRIPTION
**Pull request by JamieMagee**
###### Description of changes

- Added `nodejs-20_x`
- Set `nodejs_latest` to target `nodejs-20_x`

See https://nodejs.org/en/blog/release/v20.0.0

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by JamieMagee** [comment]
Result of `nixpkgs-review pr 226983` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>nodejs-20_x</li>
    <li>nodejs-20_x.libv8</li>
    <li>nodejs-slim-20_x</li>
    <li>nodejs-slim-20_x.libv8</li>
    <li>wasm-bindgen-cli</li>
  </ul>
</details>

File: pkgs/development/web/nodejs/v20.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+  callPackage,
+  openssl,
+  fetchpatch,
+  python3,
+  enableNpm ? true,
+}: let
```
Comment (felschr) [inline pkgs/development/web/nodejs/v20.nix:7]:
```suggestion
{ callPackage, openssl, fetchpatch, python3, enableNpm ? true }:

let
```

I think the formatting should match what's been done for the other versions.

**Comment by SuperSandro2000** [comment]
FYI https://github.com/NixOS/nixpkgs/pull/199835

**Review by felschr — CHANGES_REQUESTED** [review]

File: pkgs/development/web/nodejs/v20.nix (hunk 1 above)
Comment (winterqt) [reply pkgs/development/web/nodejs/v20.nix:7]: I'd argue it really does not matter, as we'll eventually have consistent formatting treewide. 🤷‍♀️

Comment (JamieMagee) [reply pkgs/development/web/nodejs/v20.nix]: It was applied automatically by my IDE using alejandra. I've reverted it to be inline with the other node packages. I've got my fingers crossed for RFC0101 🤞

**Comment by JamieMagee** [comment]
> FYI #199835

I've renamed the packages to `nodejs_20` and `nodejs-slim_20`.

**Review by marsam — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**